src/plugins/intel_gpu/src/graph/primitive_inst.cpp

```diff
@@ -741,8 +744,9 @@ event::ptr primitive_inst::realloc_if_needed() {
             _max_output_layout_count = _deps[0].first->_max_output_layout_count;
             GPU_DEBUG_PROFILED_STAGE_MEMALLOC_INFO("can_be_optimized");
             return ev;
-        } else if (_outputs[0] && dep_memory_ptr(0) &&
-                   _network.get_engine().is_the_same_buffer(dep_memory(0), output_memory(0))) {
+        // } else if (_outputs[0] && dep_memory_ptr(0) &&
+        //            _network.get_engine().is_the_same_buffer(dep_memory(0), output_memory(0))) {
+        } else if (_outputs[0] && dep_memory_ptr(0)) {
```

Comment:
suggestion : 
} else if (_outputs[0] && dep_memory_ptr(0) &&
                   _network.get_engine().is_the_same_buffer(dep_memory(0), output_memory(0))) {
             _outputs[0] = nullptr;
            _max_output_layout_count[0] = 0;
            // check users recursively and if the users is can_be_optimized && runtime_skippable && output_memory of user is same as current input memory, then reset the users output memory too
}

Reply:
fixed it.